src/rgw/rgw_data_sync.cc

```diff
@@ -1182,9 +1182,12 @@ class RGWDataSyncCR : public RGWCoroutine {
   }
 
   ~RGWDataSyncCR() {
+    shard_crs_lock.Lock();
```

Comment:
shouldn't need to lock in a dtor - if we get here while operate() is still running, that would indicate a separate ref counting issue. did you see this happening?